### PR TITLE
[KYUUBI #4686] Support Chose ApplicationOperator with Deploy-Mode

### LINKED ISSUE #4686: [FEATURE] ApplicationManager should add deploy-mode check when chose rmApplication
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

ApplicationManager should be referred to `Deploy-Mode` to select which applicationOperation for implementation.

### Motivation

_No response_

### Describe the solution

_No response_

### Additional context

_No response_

### Are you willing to submit PR?

- [X] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

Close #4686

ApplicationManager chose ApplicationOperator should with Deploy-Mode.

For example, Kubernetes + Client should handler by JpsApplicationOperator.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zwangsheng** [comment]
Ready for review, cc @turboFei @pan3793

**Comment by turboFei** [comment]
After this PR,

for yarn-client mode, the application info will be return by JpsApplicationOperation, it is not expected.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala [hunk 1]
```diff
@@ -225,6 +228,7 @@ object SparkProcessBuilder {
   final val TAG_KEY = "spark.yarn.tags"
   final val MASTER_KEY = "spark.master"
   final val INTERNAL_RESOURCE = "spark-internal"
+  final val DEPLOY_MODE_KEY = "spark.deploy.mode"
```
Comment (turboFei) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala:231]: spark.submit.deployMode

Comment (zwangsheng) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala]: Fixed

**Comment by zwangsheng** [comment]
close this pr to find other way

**Pull request closed without merging**